### PR TITLE
Implement %{macrobody:...} built-in for retrieving the literal macro body

### PR DESCRIPTION
**Pull request by pmatilai**
Fixes: #582

### CODE REVIEW

**Comment by pmatilai** [comment]
Some open questions:
- Should the argument be expanded first, like doFoo() does? Using the literal value has its pros but I guess its cons too.
- Is raising an error on undefined macro a reasonable thing to do? If not, what should it do on undefined macros?

I was also tempted to add %{opts:...} which would return the option declaration string (if any), but there's no direct way to distinguish between no value (ie non-parametric macro) and empty string (parametric macro which doesn't take any options) in the return value, so it'd need some other approach.

File: doc/manual/macros [hunk 1]
```diff
@@ -72,6 +72,7 @@ to perform useful operations. The current list is
 	%define ...	define a macro
 	%undefine ...	undefine a macro
 	%global ...	define a macro whose body is available in global context
+	%{body:...}	literal body of a macro
```
Comment (voxik) [inline doc/manual/macros]: I am sorry, but from the macro name neither from its description I can tell what it does. Isn't 'verbatim' more fitting name here?

Comment (pmatilai) [reply doc/manual/macros]: See the discussion in #582, the problem with %{verbatim:..} and other similar variants is that the argument is ambiguous, whereas the term "body" is unambiguous within rpm terminology, including the document in question. The description could be better, no disagreement there.

**Comment by mlschroe** [comment]
I'd prefer if we named this %macrobody, so that there's less chance that it clashes with somebody else's use of %body.

**Comment by pmatilai** [comment]
I can live with macrobody, a little bit of name-spacing never hurt anybody.

In the meanwhile, I'm starting to think it does need to expand its argument like everything else, this is too limiting. Ie it'd still take a macro name as an argument, but that the name can come from a macro instead of having to be a literal as in this version.

**Comment by mlschroe** [comment]
For the record: I also agree that it should expand the argument ;)

**Comment by pmatilai** [comment]
Renamed to %{macrobody:..} and argument expanded now.

**Comment by mlschroe** [comment]
As it now expands the argument I think it should be handled in doFoo().

**Comment by pmatilai** [comment]
Yeah it could go to doFoo(), but then I'd actually on a mission to eliminate the whole doFoo() (replaced by individual functions to do just one thing), so I'm trying to avoid adding more.
The expansion logic around g/gn could and probably should be handled centrally in the expandMacros() call-site.

**Pull request merged**